apollo-federation/src/sources/connect/validation/extended_type.rs

```diff
@@ -346,3 +425,15 @@ fn get_missing_connect_directive_message(
         locations: field.line_column_range(source_map).into_iter().collect(),
     }
 }
+
+/// Extract all seen parameters from a JSONSelection
+///
+/// TODO: this is copied from expand/mod.rs
```

Comment:
It seems like this should be fairly trivial to add to `JSONSelection` - is there a reason it can't be fixed in this PR?

Reply:
killed this function entirely